packages/tpetra/core/test/MatrixMatrix/MatrixMatrix_UnitTests.cpp

```diff
@@ -1588,7 +1593,14 @@ TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL(Tpetra_MatMat, threaded_add_sorted, SC, LO, GO
         std::string("Threaded addition of sorted Kokkos::CrsMatrix produced row ") + std::to_string(i) + " with the wrong number of entries.");
     for(size_t j = 0; j < Crowlen; j++)
     {
-      TEUCHOS_TEST_FOR_EXCEPTION(valsCRS[2](Crowstart + j) != correctVals[colindsCRS[2](Crowstart + j)], std::logic_error,
+      size_t Coffset = Crowstart + j;
+      if(j > 0)
+      {
+        //Check entries are sorted
+        TEUCHOS_TEST_FOR_EXCEPTION(colindsCRS[2](Coffset - 1) > colindsCRS[2](Coffset), std::logic_error,
+            std::string("Threaded addition of unsorted Kokkos::CrsMatrix: row ") + std::to_string(i) + " is not sorted");
```

Comment:
You don't need to cast to string here, and you can use `<<` instead of `+`.  The third macro argument gets pasted in like this:
```c++
std::ostringstream os;
os << THIRD_ARG;
```